core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -701,14 +704,24 @@ private[kafka] class Processor(val id: Int,
         openOrClosingChannel(receive.source) match {
           case Some(channel) =>
             val header = RequestHeader.parse(receive.payload)
-            val connectionId = receive.source
-            val context = new RequestContext(header, connectionId, channel.socketAddress,
-              channel.principal, listenerName, securityProtocol)
-            val req = new RequestChannel.Request(processor = id, context = context,
-              startTimeNanos = time.nanoseconds, memoryPool, receive.payload, requestChannel.metrics)
-            requestChannel.sendRequest(req)
-            selector.mute(connectionId)
-            handleChannelMuteEvent(connectionId, ChannelMuteEvent.REQUEST_RECEIVED)
+            if (header.apiKey() == ApiKeys.SASL_HANDSHAKE && channel.maybeBeginServerReauthentication(receive, time.milliseconds))
+              trace(s"Begin re-authentication: $channel")
+            else {
+              if (channel.serverAuthenticationSessionExpired(time.nanoseconds)) {
+                channel.disconnect()
+                debug(s"Disconnected expired channel: $channel")
+                expiredConnectionsKilledCount.inc()
+                throw new RuntimeException("Disconnected expired session: " + channel + "; " + header)
```

Comment:
This results in errors logged in the broker with stack trace. I don't think we want that.

Reply:
👍 Removed